File: sessions_mount_dir/9kip5xxkrzeufyu8lcpahfxk0/backend/app/routers/member_cards.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from typing import List

from ..database import get_db
from ..models import Member, MemberCard
from ..schemas import MemberCardCreate, MemberCardUpdate, MemberCardResponse
from ..enums import MemberCardStatus

router = APIRouter(
    prefix="/api/v1/members",
    tags=["Member Cards"]
)

@router.get("/{member_id}/cards", response_model=List[MemberCardResponse])
def get_member_cards(member_id: int, db: Session = Depends(get_db)):
    """
    Get all cards for a specific member
    """
    member = db.query(Member).filter(Member.id == member_id).first()
    if not member:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Member with id {member_id} not found"
        )
    
    return db.query(MemberCard).filter(MemberCard.member_id == member_id).all()

@router.post("/{member_id}/cards", response_model=MemberCardResponse, status_code=status.HTTP_201_CREATED)
def create_member_card(member_id: int, card: MemberCardCreate, db: Session = Depends(get_db)):
    """
    Create a new card for a member
    """
    member = db.query(Member).filter(Member.id == member_id).first()
    if not member:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Member with id {member_id} not found"
        )
    
    db_card = MemberCard(**card.dict(), member_id=member_id)
    db.add(db_card)
    db.commit()
    db.refresh(db_card)
    
    return db_card

@router.put("/{member_id}/cards/{card_id}", response_model=MemberCardResponse)
@router.put("/{member_id}/cards/{card_id}", response_model=MemberCardResponse)
def update_member_card(
    member_id: int,
    card_id: int,
    card_update: MemberCardUpdate,
    db: Session = Depends(get_db)
):
    """
    Update a member's card information
    """
    member = db.query(Member).filter(Member.id == member_id).first()
    if not member:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Member with id {member_id} not found"
        )

    db_card = db.query(MemberCard).filter(
        MemberCard.id == card_id,
        MemberCard.member_id == member_id
    ).first()

    if not db_card:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Card with id {card_id} not found for member {member_id}"
        )

    for field, value in card_update.dict(exclude_unset=True).items():
        setattr(db_card, field, value)

    db.commit()
    db.refresh(db_card)

    return db_card
@router.delete("/{member_id}/cards/{card_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_member_card(
    member_id: int,
    card_id: int,
    db: Session = Depends(get_db)
):
    """
    Delete a member's card
    """
    member = db.query(Member).filter(Member.id == member_id).first()
    if not member:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Member with id {member_id} not found"
        )

    db_card = db.query(MemberCard).filter(
        MemberCard.id == card_id,
        MemberCard.member_id == member_id
    ).first()

    if not db_card:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Card with id {card_id} not found for member {member_id}"
        )

    db.delete(db_card)
    db.commit()

    return None
@router.patch("/{member_id}/cards/{card_id}/status", response_model=MemberCardResponse)
def update_member_card_status(
    member_id: int,
    card_id: int,
    status: MemberCardStatus,
    db: Session = Depends(get_db)
):
    """
    Update member card status (active/inactive/expired)
    """
    member = db.query(Member).filter(Member.id == member_id).first()
    if not member:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Member with id {member_id} not found"
        )

    db_card = db.query(MemberCard).filter(
        MemberCard.id == card_id,
        MemberCard.member_id == member_id
    ).first()

    if not db_card:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Card with id {card_id} not found for member {member_id}"
        )

    db_card.status = status
    db.commit()
    db.refresh(db_card)

    return db_card
def update_member_card(
    member_id: int, 
    card_id: int, 
    card_update: MemberCardUpdate, 
    db: Session = Depends(get_db)
):
    """
    Update a member's card information
    """
    member = db.query(Member).filter(Member.id == member_id).first()
    if not member:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Member with id {member_id} not found"
        )
    
    db_card = db.query(MemberCard).filter(
        MemberCard.id == card_id,
        MemberCard.member_id == member_id
    ).first()
    
    if not db_card:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Card with id {card_id} not found for member {member_id}"
        )
    
    for field, value in card_update.dict(exclude_unset=True).items():
        setattr(db_card, field, value)
    
    db.commit()
    db.refresh(db_card)
    
    return db_card

File: sessions_mount_dir/9kip5xxkrzeufyu8lcpahfxk0/frontend/src/api/member_cards.js
import axios from 'axios';

const API_BASE_URL = '/api/v1';

/**
 * 获取会员卡列表
 * @param {number} memberId - 会员ID
 * @returns {Promise} Axios Promise
 */
export const getMemberCards = (memberId) => {
  return axios.get(`${API_BASE_URL}/members/${memberId}/cards`);
};

/**
 * 创建会员卡
 * @param {number} memberId - 会员ID
 * @param {Object} cardData - 会员卡数据
 * @param {string} cardData.card_number - 卡号
 * @param {string} cardData.expiry_date - 过期日期
 * @param {string} cardData.status - 状态
 * @returns {Promise} Axios Promise
 */
export const createMemberCard = (memberId, cardData) => {
  return axios.post(`${API_BASE_URL}/members/${memberId}/cards`, cardData);
};

/**
 * 更新会员卡信息
 * @param {number} memberId - 会员ID
 * @param {number} cardId - 会员卡ID
 * @param {Object} cardData - 更新的会员卡数据
 * @returns {Promise} Axios Promise
 */
export const updateMemberCard = (memberId, cardId, cardData) => {
  return axios.put(`${API_BASE_URL}/members/${memberId}/cards/${cardId}`, cardData);
};

/**
 * 更新会员卡状态
 * @param {number} memberId - 会员ID
 * @param {number} cardId - 会员卡ID
 * @param {string} status - 新状态
 * @returns {Promise} Axios Promise
 */
export const updateCardStatus = (memberId, cardId, status) => {
  return axios.patch(`${API_BASE_URL}/members/${memberId}/cards/${cardId}/status`, { status });
};

/**
 * 获取所有会员卡（管理员用）
 * @returns {Promise} Axios Promise
 */
export const getAllMemberCards = () => {
  return axios.get(`${API_BASE_URL}/members/cards`);
};

export default {
  getMemberCards,
  createMemberCard,
  updateMemberCard,
  updateCardStatus,
  getAllMemberCards
};